CI/CD enforcement documentation (#2080)
astronomer/docs: astro/configure-deployment-resources.md

In addition to alert emails for your Deployments, Astronomer recommends subscribing to the [Astro status page](https://status.astronomer.io). When you subscribe, you'll receive email notifications about system-wide incidents in real time.

## Enforce CI/CD deploys

> By default, Deployments accept code deploys from any authenticated source. When you enforce CI/CD deploys for a Deployment, the Deployment will accept code deploys only if the deploy was triggered using a Deployment API key or Workspace token. 

Reviewer: ```suggestion
By default, Deployments accept code deploys from any authenticated source. When you enforce CI/CD deploys for a Deployment, the Deployment only accepts code deploys if they are triggered with a Deployment API key or Workspace token. 
```
